### DISCUSSION TITLE
[React Router v6 - Zustand - Electron] Store is undefined outside of react component

### DISCUSSION THREAD

**Discussion by Shackey666** [pmndrs/zustand#2167] [Q&A]
Hello Zustand community!  

I have an issue regarding Zustand and my router: 

I have my App:

`const App = () => {
  const store = useLauncherStore.getState();
  console.log('APP 0', store);

  return (
    <React.StrictMode>
        <QueryClientProvider client={queryClient}>
              <RouterProvider router={router} fallbackElement={<AppLoader />} />
        </QueryClientProvider>
    </React.StrictMode>
  );
};`

This is my Router:

`export const router = createHashRouter([
  {
    element: <AppLayout />,
    loader:appLayoutLoader,
    children: [
      {
        index: true,
        path: ROUTES.HOME,
        loader: homeLoader,
      },
    ]
  }
])`

This is my AppLayoutLoader:

`
import { useStore } from '../../store';

export const appLayoutLoader = async () => {
const store = useStore.getState();

  try {
    await Promise.all([
      await new Promise((r) => setTimeout(r, configuration.appLayoutLoaderMinimumDuration)),
      (async () => {
        await store.application.actions.initializeApplication();
        await store.library.actions.initializeLibrary();
      })(),
    ]);

    // if user is logged-in redirect it to game page
    if (useLauncherStore.getState().application.actions.isLoggedIn()) {
      throw redirect(ROUTES.GAME_PAGE);
    }

    throw redirect(ROUTES.HOME);
  } finally {
    return null;
  }
};`

My store is welld display with the console.log in App.tsx but not in the AppLayoutloader. 
I have this error in the loader: cannot read property of getState() of undefined.
What am I doing wrong?

**Comment by dbritto-dev** [comment]
@Shackey666 would you mind creating a codesandbox repro?

**Reply by Shackey666** [reply]
Fixed it sorry, it was a router issue. My bad sorry, here is a sandbox just in case. => https://codesandbox.io/s/react-router-dom-loaders-layout-createbrowserrouter-zustand-g7f2x5
